db/write_batch.cc

```diff
@@ -1222,6 +1223,10 @@ class MemTableInserter : public WriteBatch::Handler {
   DupDetector       duplicate_detector_;
   bool              dup_dectector_on_;
 
+  bool hint_per_batch_;
+  // Hints for this batch
+  std::unordered_map<MemTable*, void*> hint_;
```

Comment:
We ever caused regression on Windows when introducing post map. The reason is that on Windows, those STL containers invoke a malloc even without inserting anything. That's why there is a very odd logic GetPostMap(). Just see the line blow. I'm not sure whether things have improved on Windows. To be safe, we can do exactly the same as mem_post_info_map_.